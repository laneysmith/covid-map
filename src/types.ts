export const COLORS = [
    '#DEEDCF',
    '#B8DEAA',
    '#88CE86',
    '#62BE73',
    '#3FAC6C',
    '#1D9A6C',
    '#188977',
    '#137177',
    '#0E4D64',
    '#0A2F51',
];

export enum Variable {
    CASES = 'cases',
    DEATHS = 'deaths',
}

export type ColorScales = { [key in Variable]: [] };

export type FipsStats = { [key in Variable]: number }

export type ResponseData = { data: object, maxCases: number, maxDeaths: number }
