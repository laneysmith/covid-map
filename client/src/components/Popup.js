import React from 'react';

import { formatNumber } from '../utils';

const Popup = ({ feature }) => {
  const { id, COUNTY: county } = feature.properties;
  const { cases, deaths } = feature.state;

  return (
    <div id={`popup-${id}`}>
      <h3>{county}</h3>
      {typeof cases === 'number' && (
        <div>
          <b>Cases:</b> {formatNumber(cases)}
        </div>
      )}
      {typeof deaths === 'number' && (
        <div>
          <b>Deaths:</b> {formatNumber(deaths)}
        </div>
      )}
    </div>
  );
};

export default Popup;
