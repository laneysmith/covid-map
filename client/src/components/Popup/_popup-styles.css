.mapboxgl-popup-content {
  padding: 10px;
  min-width: 150px;
}

.mapboxgl-popup-content h3 {
  margin: 0 0 6px 0;
  padding-bottom: 6px;
  border-bottom: 1px solid #efefef;
}
