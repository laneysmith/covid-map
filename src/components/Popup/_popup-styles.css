.mapboxgl-popup-content {
  padding: 10px;
  min-width: 150px;
}

.mapboxgl-popup-content h3 {
  margin: 0 0 6px 0;
  padding-bottom: 6px;
  border-bottom: 1px solid #efefef;
}

.exception-note {
  color: grey;
  font-size: 8pt;
  line-height: 10pt;
}

.animation-control-icon {
  height: 12px;
  width: 12px;
}