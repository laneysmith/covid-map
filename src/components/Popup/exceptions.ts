const NYC_EXCEPTION_NOTE =
  'Data for New York, Kings, Queens, Bronx and Richmond counties were reported in aggregate.';
const ALAMEDA_EXCEPTION_NOTE =
  'Counts for Alameda County include cases and deaths from Berkeley and the Grand Princess cruise ship.';
const DOUGLAS_EXCEPTION_NOTE =
  'Counts for Douglas County include cases brought to the state from the Diamond Princess cruise ship.';

export const EXCEPTIONS: Map<string, string> = new Map([
  ['36005', NYC_EXCEPTION_NOTE], // Bronx County
  ['36047', NYC_EXCEPTION_NOTE], // Kings County
  ['36061', NYC_EXCEPTION_NOTE], // New York County
  ['36081', NYC_EXCEPTION_NOTE], // Queens County
  ['36085', NYC_EXCEPTION_NOTE], // Richmond County
  ['06001', ALAMEDA_EXCEPTION_NOTE], // Alameda County, CA
  ['31055', DOUGLAS_EXCEPTION_NOTE], // Douglas County, NE
]);