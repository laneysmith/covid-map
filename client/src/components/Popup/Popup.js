import React from 'react';

import { formatNumber } from '../../utils';
import { EXCEPTIONS } from './exceptions';
import './_popup-styles.css';

const Popup = ({ feature }) => {
  const { id, properties } = feature;
  const { COUNTY: county } = properties;
  const { cases, deaths } = feature.state;
  const hasCasesData = typeof cases === 'number';
  const hasDeathsData = typeof deaths === 'number';
  const stringId = id.toString();
  const exceptionNote = EXCEPTIONS[stringId] ? `Note: ${EXCEPTIONS[stringId]}` : null;

  return (
    <div id={`popup-${id}`}>
      <h3>{county}</h3>
      {hasCasesData && (
        <div>
          <b>Cases:</b> {formatNumber(cases)}
        </div>
      )}
      {hasDeathsData && (
        <div>
          <b>Deaths:</b> {formatNumber(deaths)}
        </div>
      )}
      {!hasCasesData && !hasDeathsData && (
        <div>
          <i>No data for this county.</i>
        </div>
      )}
      {exceptionNote && <div className="exception-note">{exceptionNote}</div>}
    </div>
  );
};

export default Popup;
