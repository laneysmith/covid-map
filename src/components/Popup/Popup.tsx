import React, { HTMLAttributes } from 'react';
// import { MapboxGeoJSONFeature } from 'mapbox-gl';

import { formatNumber } from '../../utils';
import { EXCEPTIONS } from './exceptions';
import './_popup-styles.css';

interface PopupProps extends HTMLAttributes<HTMLElement> {
  // TODO: not 'any'; reconcile MapboxGeoJSONFeature type & IFeature interface
  feature: any;
}

const Popup: React.SFC<PopupProps> = ({ feature }) => {
  const { id, properties, state } = feature;
  const { COUNTY: county } = properties;
  const { cases, deaths } = state;
  const hasCasesData = typeof cases === 'number';
  const hasDeathsData = typeof deaths === 'number';
  const stringId = id?.toString();
  const exceptionNote = EXCEPTIONS.get(stringId) ? `Note: ${EXCEPTIONS.get(stringId)}` : null;

  return (
    <div id={`popup-${id}`}>
      <h3>{county}</h3>
      {hasCasesData && (
        <div>
          <b>Cases:</b> {formatNumber(cases)}
        </div>
      )}
      {hasDeathsData && (
        <div>
          <b>Deaths:</b> {formatNumber(deaths)}
        </div>
      )}
      {!hasCasesData && !hasDeathsData && (
        <div>
          <i>No data for this county.</i>
        </div>
      )}
      {exceptionNote && <div className="exception-note">{exceptionNote}</div>}
    </div>
  );
};

export default Popup;
