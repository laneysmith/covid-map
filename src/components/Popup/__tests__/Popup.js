import React from 'react';
import { render } from '@testing-library/react';

import Popup from '../Popup';
import { Variable } from '../../../types';

describe('Popup', () => {
  it('renders correctly with exception note', () => {
    const feature = {
      id: '36005',
      properties: {
        COUNTY: 'Bronx County',
      },
      state: {
        cases: 12,
        deaths: 1,
      },
    };
    const { container } = render(<Popup feature={feature} />);

    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders correctly without exception note', () => {
    const feature = {
      id: '53061',
      properties: {
        COUNTY: 'Bronx County',
      },
      state: {
        cases: 234,
        deaths: 9,
      },
    };
    const { container } = render(<Popup feature={feature} />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
