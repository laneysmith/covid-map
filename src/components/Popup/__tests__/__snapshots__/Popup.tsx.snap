// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Popup renders correctly with exception note 1`] = `
<div
  id="popup-36005"
>
  <h3>
    Bronx County
  </h3>
  <div>
    <b>
      Cases:
    </b>
     
    12
  </div>
  <div>
    <b>
      Deaths:
    </b>
     
    1
  </div>
  <div
    class="exception-note"
  >
    Note: Data for New York, Kings, Queens, Bronx and Richmond counties were reported in aggregate.
  </div>
</div>
`;

exports[`Popup renders correctly without exception note 1`] = `
<div
  id="popup-53061"
>
  <h3>
    Bronx County
  </h3>
  <div>
    <b>
      Cases:
    </b>
     
    234
  </div>
  <div>
    <b>
      Deaths:
    </b>
     
    9
  </div>
</div>
`;
